dotCMS/src/main/java/com/dotmarketing/portlets/templates/action/EditTemplateAction.java

```diff
@@ -292,10 +292,25 @@ else if ((cmd != null) && cmd.equals(com.dotmarketing.util.Constants.FULL_DELETE
 			{
 				Logger.debug(this,"Calling Full Delete Method");
 				WebAsset webAsset = (WebAsset) req.getAttribute(WebKeys.TEMPLATE_EDIT);
-				if(WebAssetFactory.deleteAsset(webAsset,user)) {
-					SessionMessages.add(httpReq, "message", "message." + webAsset.getType() + ".full_delete");
+
+				List<Contentlet> pages = APILocator.getContentletAPI().findPagesByTemplate((Template)webAsset, user, false);
+				if (pages != null && !pages.isEmpty()) {
+
+					StringBuilder error = new StringBuilder();
+					error.append(LanguageUtil.get(user, "message.template.full_delete.error")).append(" ");
+					error.append(webAsset.getName()).append("<br>");
+
+					for (Contentlet page : pages) {
+						error.append("- ").append(page.getTitle()).append("<br>");
+					}
+					SessionMessages.add(httpReq,"error", error.toString());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)